### PR TITLE
Replace text-encoding package with fastestsmallesttextencoderdecoder

### PR DESCRIPTION
**Pull request by adroitwhiz**
### Proposed Changes

This PR replaces the [deprecated and unmaintained `text-encoding` package](https://www.npmjs.com/package/text-encoding), which polyfills the `TextEncoder` and `TextDecoder` APIs, with [`fastestsmallesttextencoderdecoder`](https://www.npmjs.com/package/fastestsmallesttextencoderdecoder).

### Reason for Changes

The main draw of `text-encoding` is that it supports text encodings other than UTF-8. However, we never use this functionality.

This entirely unused support for legacy encodings adds **500kb** of encoding data to the package. Replacing the polyfill with `fastestsmallesttextencoderdecoder`, which only supports UTF-8, removes this 500kb.

The `text-encoding` package is also pulled in by a few other Scratch repositories (scratch-vm, scratch-gui, and scratch-sb1-converter), and they don't use the non-UTF-8 functionality either. I'd remove it in all those repositories too, but I'd rather not "spam" pull requests.

I have not benchmarked load times. That may be worth looking into.

### Test Coverage

I have tested this in Edge, which is the only browser without native support for the Text Encoding API, and it works.

### CODE REVIEW

**Comment by cwillisf** [comment]
What problems do you think this could cause if it doesn't work? It seems like this would primarily impact loading and saving of text assets, like project JSON for example, so maybe loading and saving projects on the web site and/or SB3 files should be tested in particular. Does that sound right to you?
TL;DR: Could you add testing steps to this?

**Review by cwillisf — APPROVED** [review]
Code looks good! Could you add testing info before merging? Thanks!

**Comment by fsih** [comment]
Usages of text encoder/decoder:
- to fix quirks in .sb2 svgs to display properly in scratch 3
- to load all SVGs
- to inline fonts in SVGs to display correctly in thumbnails

**Comment by fsih** [comment]
Testing: load an SB2 with SVG assets, and make sure that quirks are fixed correctly (text fonts are updated, viewbox is not messed up)
Make sure that the font is showing up in thumbnails
